Presentation.Web/Controllers/API/V1/UserController.cs

```diff
@@ -214,6 +218,29 @@ public HttpResponseMessage GetUsersWithCrossAccess()
                 .Match(Ok, FromOperationError);
         }
 
+        /// <summary>
+        /// Returns the users with matching name or email
+        /// </summary>
+        /// <param name="nameOrEmailQuery">Name or email of the user</param>
+        /// <returns>A list of users</returns>
+        [HttpGet]
+        [Route("api/users/search/{nameOrEmailQuery}")]
+        public HttpResponseMessage SearchUsers(string nameOrEmailQuery, [FromUri] BoundedPaginationQuery paginationQuery = null)
```

Comment:
USer query parameters and not route parameters for this. Remember to use url encoding and url decoding.